ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/atomicaction/AtomicActionAsyncTest.java

```diff
@@ -49,6 +56,28 @@ public void testAbort() throws Exception {
     }
 
     @Test
+    public void testAbortAsyncAC() throws Exception {
+        AtomicActionTestBase.init(false);
+        arjPropertyManager.getCoordinatorEnvironmentBean().setAsyncAfterSynchronization(true);
+
+        final Thread thread = Thread.currentThread();
+        executeTest(true, ActionStatus.COMMITTED, new SyncRecord[] { new SyncRecord() {
+            @Override
+            public boolean afterCompletion(int status) {
+                testAbortAsyncAC = testAbortAsyncAC && (Thread.currentThread() != thread);
```

Comment:
I don't believe it was designed with the executor to do that. the test is running in main thread and an executor pool executes the syncs